app/queries/get_incomplete_reference_applications_ready_to_nudge.rb

```diff
@@ -1,49 +1,35 @@
 class GetIncompleteReferenceApplicationsReadyToNudge
-  MAILER = 'candidate_mailer'.freeze
-  MAIL_TEMPLATE = 'nudge_unsubmitted_with_incomplete_references'.freeze
-  COMMON_COMPLETION_ATTRS = GetUnsubmittedApplicationsReadyToNudge::COMMON_COMPLETION_ATTRS
+  COMMON_COMPLETION_ATTRS_WITHOUT_REFERENCES = GetUnsubmittedApplicationsReadyToNudge::COMMON_COMPLETION_ATTRS.filter do |attr|
+    attr != 'references_completed'
+  end
 
   def call
     uk_and_irish_names = NATIONALITIES.select do |code, _name|
       code.in?(ApplicationForm::BRITISH_OR_IRISH_NATIONALITIES)
     end.map(&:second)
-    uk_and_irish = uk_and_irish_names.map { |name| ActiveRecord::Base.connection.quote(name) }.join(',')
 
     ApplicationForm
+      # Only candidates with some unsubmitted choices
+      .joins(:application_choices)
       .joins(:candidate)
-      .where.not('candidate.submission_blocked': true)
-      .where.not('candidate.account_locked': true)
-      .where.not('candidate.unsubscribed_from_emails': true)
+      .joins("LEFT OUTER JOIN emails ON emails.application_form_id = application_forms.id AND emails.mailer = 'candidate_mailer' AND emails.mail_template = 'nudge_unsubmitted_with_incomplete_references'")
+      .joins('LEFT OUTER JOIN application_choices ac_primary ON ac_primary.application_form_id = application_forms.id')
+      .joins('LEFT OUTER JOIN course_options ON course_options.id = ac_primary.course_option_id')
+      .joins("LEFT OUTER JOIN courses courses_primary ON courses_primary.id = course_options.course_id AND LOWER(courses_primary.level) = 'primary'")
       .current_cycle
+      # Only applications forms with at least one unsubmitted application choice (inflight)
+      .where('application_choices.status': 'unsubmitted')
```

Comment:
Two small non blocking questions. 

1. If the candidates fill all their sections and leave the references section blank AND doesn't have any application choices - are these expected to receive the nudge? If so can we add a spec for this? (It is non blocking but it might contain a bug if we count only people with application choices.
2. If the above is true we should send to people with no applications what happens in the period find cloes <-> apply opens? I assume same behaviour so maybe adding one more spec for this?

They are non blocking questions

Reply:
1. They would not receive a nudge. This is actually tested in the specs. With the inner join on application_choices, anyone without application choices will not receive a nudge.
3. They can start making choices when find opens, they just can't submit them. So between apply deadline (17 September) and find reopens (1 October) no one new would qualify to receive this email.